File: Ventrue/src/core/Effect/EffectCmd/EffectEqualizerCmd.cpp
#include"EffectEqualizerCmd.h"

namespace ventrue
{
	void EffectEqualizerCmd::SetFreqBandGain(int bandIdx, float gainDB)
	{
		EffectTask* task = new EffectTask(effect);
		task->processCallBack = _SetFreqBandGain;
		task->valuei[0] = bandIdx;
		task->valuef[0] = gainDB;
		ventrue->PostTask(task);
	}

	void EffectEqualizerCmd::_SetFreqBandGain(Task* task)
	{
		EffectTask* effectTask = (EffectTask*)task;
		Equalizer& eq = *(Equalizer*)(effectTask->effect);
		eq.SetFreqBandGain(effectTask->valuei[0], effectTask->valuef[0]);
	}

}

File: Ventrue/andstudio/Ventrue/app/src/main/java/cymheart/ventrue/effect/EffectEqualizerCmd.java
package cymheart.ventrue.effect;

import cymheart.ventrue.Ventrue;

public class EffectEqualizerCmd {

    private Ventrue ventrue;
    private Equalizer eq;

    public EffectEqualizerCmd(Ventrue ventrue, Equalizer eq)
    {
        this.ventrue = ventrue;
        this.eq = eq;
        ndkEffectEqCmd = ndkCreateEffectEqualizerCmd(ventrue.GetNdkVentrue(), eq.GetNdkEffect());
    }


    public void SetFreqBandGain(int bandIdx, float gainDB)
    {
        ndkSetFreqBandGain(ndkEffectEqCmd, bandIdx, gainDB);
    }


    private long ndkEffectEqCmd;
    private static native long ndkCreateEffectEqualizerCmd(long ndkVentrue, long ndkEqualizer);
    private static native void ndkSetFreqBandGain(long ndkEffectEqCmd, int bandIdx, float gainDB);
}
